### ISSUE TITLE
Dynamic classes cause PicklingError 

### ISSUE THREAD

**Issue by louisabraham** [EpistasisLab/tpot#781]
I tried to pickle a TPOTRegressor and had this Exception:

    PicklingError: Can't pickle <class 'tpot.operator_utils.RandomForestRegressor__bootstrap'>: attribute lookup RandomForestRegressor__bootstrap on tpot.operator_utils failed

It seems pickle has issues with dynamically created classes, probably from [TPOTOperatorClassFactory](https://github.com/EpistasisLab/tpot/blob/master/tpot/operator_utils.py#L130) and [ARGTypeClassFactory](https://github.com/EpistasisLab/tpot/blob/master/tpot/operator_utils.py#L109).

According to the documentation of pickle, it is sufficient to write a [__reduce__](https://docs.python.org/3/library/pickle.html#object.__reduce__) method that reconstructs the object, which is not very easy.

Another possibility would be not to use those dynamic classes. In the past I have done similar things, and in the end dynamic classes have always been been a bad idea.

The TPOTOperatorClassFactory seems a bit complicated, but could you explain to me why a ARGTypeClassFactory is used?

**Comment by weixuanfu** [comment]
We need those dynamic classes to build strong type via deap (see [here](https://deap.readthedocs.io/en/master/examples/gp_spambase.html)) based on default/customized configuration of operators and parameters (see [here](https://epistasislab.github.io/tpot/using/#customizing-tpots-operators-and-parameters)).

**Comment by raybellwaves** [comment]
I also came across this. ~~Is there any guidance on pickling tpot objects?~~
Seems https://github.com/EpistasisLab/tpot/issues/11#issuecomment-341421022 gives a workaround

**Comment by weixuanfu** [comment]
Yep, that is a workaround. So far we did not have a good solution to replace those dynamic classes.
